src/org/joml/Rectanglei.java

```diff
@@ -132,20 +195,52 @@ public boolean contains(Vector2ic point) {
 
     /**
      * Check if this rectangle contains the given point <code>(x, y)</code>.
-     * 
+     *
      * @param x
      *          the x coordinate of the point to check
      * @param y
      *          the y coordinate of the point to check
      * @return <code>true</code> iff this rectangle contains the point; <code>false</code> otherwise
      */
     public boolean contains(int x, int y) {
-        return x >= minX && y >= minX && x < maxX && y < maxY;
+        return x >= minX && x <= maxX && y >= minY && y <= maxY;
     }
 
+    /**
+     * finds the rectangle that is contained between two overlapping other rectangles. if neither intersect then <code>dest</code> rectangle will be invalid.
+     * @param other
+     *          rectangle to intersect with
+     * @param dest
+     *      will hold the result
+     * @return dest
+     */
+    public Rectanglei intersect(Rectanglei other, Rectanglei dest) {
+        dest.minX = Math.max(this.minX, other.minX);
+        dest.maxX = Math.min(this.maxX, other.maxX);
+        dest.minY = Math.max(this.minY, other.minY);
+        dest.maxY = Math.min(this.maxY, other.maxY);
+        return dest;
+    }
+
+    public boolean overlaps(Rectanglei other) {
```

Comment:
Why have you deleted the original `boolean intersects(Rectanglei)` method? That's an unnecessary breaking change. It's still preferable to know whether two rectangles intersect without computing the rectangle of intersection.
If I'm not mistaken, what was previously `boolean intersects(Rectanglei)` is now `overlaps`? However I think this is too complex compared to the old `boolean intersects(Rectanglei)`.